wireguard-go-rust-wrapper/wireguard-go/platform_specific_windows.go

```diff
@@ -73,7 +74,7 @@ func (b Binder) Open(port uint16) (fns []conn.ReceiveFunc, actualPort uint16, er
 		return recv_fns, actual_port, err
 	}
 
-	recv_fns_local, actual_port, err := b.local_bind.(*WinRingBind).OpenOnLocalhost(port)
+	recv_fns_local, actual_port, err := b.local_bind.OpenOnLocalhost(port)
```

Comment:
I am wondering if this can be simplified without copy-pasting in the code from the upstream. A check can be made in Go at runtime to see if `local_bind` is `WinRingBind`, assuming it is not then it's an errorneous case where the initialization has failed and thus we could handle it(instead of crashing)?

Reply:
But we want it to work in case of failed initialization. Not just fail gracefully. That's why OpenOnLocalhost is added to `StdNetBind`. And in order to add this implementation we need a copy of it unfortunately.

Reply:
I guess it makes sense then, if it still wants to do `OpenOnLocalhost`  then these changes make sense.

+1.0